app/questionnaire/path_finder.py

```diff
@@ -53,16 +47,19 @@ def routing_path(
                     self.answer_store,
                     self.list_store,
                     current_location=current_location,
-                    routing_path=path,
                 ):
                     continue
 
             blocks.extend(group["blocks"])
 
         if blocks:
-            path = self._build_path(blocks, path, current_location)
+            routing_path_block_ids = self._build_routing_path_ids(
+                blocks, current_location
+            )
+
+        list_name = self._get_list_name(current_location)
 
-        return RoutingPath(path)
+        return RoutingPath(routing_path_block_ids, section_id, list_item_id, list_name)
```

Comment:
routing_path_block_ids will be unassigned here if there are no blocks.

Reply:
Yep - _build_routing_path_ids and RoutingPath should handle empty iterables to negate the `if` statement e.g. an empty iterable should return an emtpy RoutingPath.

Reply:
Discussed offline - making methods/class instantiation better handle empty iterables should be a separate piece of work.